com.unity.perception/Runtime/GroundTruth/Labeling/Labeling.cs

```diff
@@ -34,14 +35,19 @@ public class Labeling : MonoBehaviour
         /// </summary>
         public uint instanceId { get; private set; }
 
-        void Awake()
+        void OnDestroy()
         {
-            labelManager.Register(this);
+            labelManager.Unregister(this);
         }
 
-        void OnDestroy()
+        void OnEnable()
         {
-            labelManager.Unregister(this);
+            RefreshLabeling();
+        }
+
+        void OnDisable()
+        {
+            RefreshLabeling();
```

Comment:
Unless I'm mistaken, shouldn't we be unregistering the label on disable here vs. refreshing?

Reply:
In order for the label to be effectively removed from the ground truth generators, it needs to go through the "registration" process again, which makes it go through SetupMaterialProperties again. This is were the enabled state is checked and if disabled, the ground truth generators are reset for the object.

I think about it like this: Every labeling component that exists is registered, similar to the entity query we had. But those that are disabled are ignored in ground truth generation.

Reply:
gotcha